### PR TITLE
Feature/subtitle plugin

### PR DESCRIPTION
**Pull request by jfboeve**
Fix visibility property issue
Added viewport width and height getters and setters and incorporated into calculations
Added functions to index file

### CODE REVIEW

File: src/Subtitles/SubtitleComponent.js [hunk 1]
```diff
@@ -137,11 +153,12 @@ export default class SubtitleComponent extends Lightning.Component {
 
   _defineBreakpoint(text, breakpoint) {
     if (breakpoint >= this.maxWidth) return this.maxWidth
+    const textureDefaults = new Lightning.textures.TextTexture(this.stage).cloneArgs()
```
Comment (michielvandergeest) [inline src/Subtitles/SubtitleComponent.js:156]: I agree it seems more logical to put this functionality in the method that uses it, but I'd say it's more efficient to keep this line in the `_init()` method so it's only executed once (and not every time the breakpoint is calculated - which happens multiple times)

File: src/Subtitles/SubtitleComponent.js [hunk 2]
```diff
@@ -162,9 +179,7 @@ export default class SubtitleComponent extends Lightning.Component {
 
       this.tag('Text').text.wordWrapWidth = breakpoint
       this.tag('Text').text = v
-      setTimeout(() => {
-        this.setSmooth('alpha', 1, { duration: 0.2 })
-      }, 100)
+      this.alpha = 1
```
Comment (michielvandergeest) [inline src/Subtitles/SubtitleComponent.js:182]: 👍 as discussed .. not need to have an animation here.

File: src/Subtitles/index.js [hunk 3]
```diff
@@ -75,6 +76,27 @@ export default {
...
+    subtitles.viewportH = h
+  },
+  xPos(v) {
+    const subtitles = getOrCreateSubtitlesComponent()
+    subtitles.xPos = v
+  },
+  yPos(v) {
+    const subtitles = getOrCreateSubtitlesComponent()
+    subtitles.yPos = v
+  },
```
Comment (michielvandergeest) [inline src/Subtitles/index.js:99]: wouldn't it be more intuitive to have only one method for this that accepts an object with 4 keys (width, height, x and y)?

**Review by michielvandergeest — CHANGES_REQUESTED** [review]
see inline comments

File: src/Subtitles/SubtitleComponent.js (hunk 1 above)
Comment (jfboeve) [reply src/Subtitles/SubtitleComponent.js:156]: Oh right, forgot to revert that back!

File: src/Subtitles/index.js (hunk 3 above)
Comment (jfboeve) [reply src/Subtitles/index.js:99]: From a frontend point of view I really like the individual value method, this allows you to change the values a bit more dynamically during run time without you having to know f.e. the width of the viewport. While using methods with more parameters are handy during the initialisation phase of the player.
The individual value methods are also very handy when using the `styles` method, this allows you to do: 

Subtitles.styles({
  xPos: 'center',
  fontSize: 40,
  viewportH: 540
})

**Pull request closed without merging**